### ISSUE TITLE
[UI] Changing the Error div 

### ISSUE THREAD

**Issue by okay-pa1** [VigneshDevHub/CampX#38]
**Description**
The error div displays the error.stack which didn't seem necessary for a user.
I would like to add a home button which redirects the user to the home page and make the div a little more attractive instead.

**Comment by Vignesh025** [comment]
@okay-pa1 Your implementation looks solid, great job! However, I'm not entirely satisfied with the current UI. Would you be able to work on improving the UI? If not, I can raise a new issue and assign it to someone else. Let me know what works best for you. Thank you!

**Comment by okay-pa1** [comment]
@Vignesh025 Thanks for the feedback. 
I would like to give another shot at making it better. If u could give me a few details on what you are expecting I will take that into consideration and work again. Please let me know.

**Comment by Vignesh025** [comment]
Hey @okay-pa1 ,

I’ve just realized that the reason I kept the error stack on the screen was because we don’t have any error codes in the project yet. This way, users can get a better understanding of what went wrong using the stack trace.

If you’d like to remove the stack, we would need to add proper error codes first. Apologies for not mentioning this earlier. Let me know your thoughts!

**Comment by okay-pa1** [comment]
No Right. That totally makes sense. Correct me if I'm wrong so each route(controllers) should be revised and made sure that errors(with error code and respective error messages) are added wherever necessary. I would be really glad to work on that. Please let me know your thoughts.

**Comment by Vignesh025** [comment]
Yes @okay-pa1 , that's correct! Each route and controller should be reviewed to ensure that errors, along with their corresponding error codes and messages, are properly implemented. I appreciate your willingness to work on this. Let me know if you need any assistance!

**Comment by okay-pa1** [comment]
@Vignesh025 I'll let you know if I need any assistance as I work on it. Thank you.

**Comment by Vignesh025** [comment]
Hey @okay-pa1 , are you still working on this issue? If not, please let me know so I can assign it to someone else. Thanks!

**Comment by okay-pa1** [comment]
Hey @Vignesh025 I am still working on the issue. I'm sorry for the delay since I had some other academic work. I will raise a PR today for a review. I hope you can give me another day's time. Thank you.

**Comment by Vignesh025** [comment]
Hey, no worries! I was just making sure everyone is still working on their assigned tasks. Take your time, and feel free to submit the PR when you're ready. Thanks for the update!

**Comment by okay-pa1** [comment]
Hey @Vignesh025 ! Sorry I took a lot of time to just come up with this. This is the solution that I thought if after going through all the routes. The errors that we could handle or implement so that it pops into that error div would be:

- 403(Forbidden): In the isAuthor middleware function req.flash() has been used to put up a simple message "You do not have permission to do that!" where i thought a 403 error can be shown instead. This would happen when an authenticated person tries to edit an authorized campground.
- 400(Bad Request): Suppose a person is trying to view a campground through an invalid ObjectId.
- 405(Method not allowed): This seems unnecessary but please let me know your thoughts, if a user tries to access /reviews/ or /reviews/:id it gives a 404 error.
- 502(Bad Gateway): This can be shown if there's any issue with mapbox while creating a campground.
- 429(Too many requests): Please let me know if I should be handling rate-limiting.

The rest of the routes seemed to be perfect and I didn't want to meddle with it. Please let me know your thoughts on what I can do.

**Comment by Vignesh025** [comment]
@okay-pa1 , You can proceed with opening a PR.
